src/DocScan/Session/Retrieve/CheckResponse.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yoti\DocScan\Session\Retrieve;
+
+use Yoti\Exception\DateTimeException;
+use Yoti\Util\DateTime;
+
+class CheckResponse
+{
+
+    /**
+     * @var string|null
+     */
+    private $id;
+
+    /**
+     * @var string|null
+     */
+    private $state;
+
+    /**
+     * @var ReportResponse|null
+     */
+    private $report;
+
+    /**
+     * @var \DateTime|null
+     */
+    private $created;
+
+    /**
+     * @var \DateTime|null
+     */
+    private $lastUpdated;
+
+    /**
+     * @var string[]
+     */
+    private $resourcesUsed = [];
+
+    /**
+     * @var GeneratedMedia[]
+     */
+    private $generatedMedia = [];
+
+    /**
+     * CheckResponse constructor.
+     * @param array<string, mixed> $checkData
+     * @throws DateTimeException
+     */
+    public function __construct(array $checkData)
+    {
+        $this->id = $checkData['id'] ?? null;
+        $this->state = $checkData['state'] ?? null;
+        $this->resourcesUsed = $checkData['resources_used'] ?? [];
+
+        if (isset($checkData['generated_media'])) {
+            $this->generatedMedia = $this->parseGeneratedMedia($checkData['generated_media']);
```

Comment:
You could replace the private method with this _(just a suggestion, fine to leave this as is)_

```suggestion
        foreach ($checkData['generated_media'] as $generatedMedia) {
            $this->generatedMedia[] = new GeneratedMedia($generatedMedia);
        }
```

Reply:
Fixed in 93f3254